### PR TITLE
Audio Recording extension reference pages

### LINKED ISSUE #5222: Need to add the help doc of extension 'Record'
**Describe the bug**
Need to add the help doc of extension '**Record**'

**Steps to reproduce the behavior**
1. Navigate to https://makecode.microbit.org/beta#
2. Create a project
3. Add extension '**Record**'
4. Right-click on any block of '**Record**' and choose '**Help**'

**Actual behavior**
No help doc of extension '**Record**'

**Expect behavior**
Need to add the help doc of extension '**Record**'

**Screenshot**

**Additional context**
1.OS: Windows11
2.makecode.microbit.org version: 5.1.37
3.Microsoft MakeCode version: 8.6.48

### LINKED ISSUE #5240: Docs for the recording extension
Right now, the recording extension's "Learn More" page will give users a 404; we want users to be able to visit that docs page. 

I'm not a huge fan of how I wrote these, so please modify, but I wanted to give a bit more context on what each block does:

**Record block:** records user audio for a short period of time. Recording "until done" will wait until the audio buffer is full before executing any code after it. Until done can still be interrupted by events, though! Recording "in the background" will start recording and will continue to fill the audio buffer but will also continue executing sequential code as the buffer fills. This means if you have record in background right before play, you won't hear any sound because the audio immediately stopped recording!
 
**Play block:** plays whatever recorded audio is in the audio buffer. If there is nothing in the buffer, nothing will happen. Playing "until done" and "in the background" behaves similarly as the record block.
 
**Audio is block:** checks the state of the audio buffer. Will return true or false to tell the user if the audio is playing, recording, stopped or has nothing in it. 
 
**Set sample rate block:** modifies the speed at which the audio buffer collects or plays back data. Modifying this up and down for playback or recording will make your recording sound fast or slow!
 
**Set microphone sensitivity block:** modifies how much sound the microphone on the micro:bit will pick up. Setting the microphone to low sensitivity will make the microphone pick up louder sounds. Setting the microphone to high sensitivity will make the microphone pick up all sorts of noise that you might not hear!

### PR DESCRIPTION
**Pull request by ganicke**
Add a base set of reference pages and a top level reference page to satisfy the "Learn more" link destination.

Closes #5222, closes #5240

### CODE REVIEW

**Comment by srietkerk** [comment]
Oh my gosh, I am so sorry that I'm just getting to this. This just completely slipped through the cracks for me.

File: libs/audio-recording/docs/reference/record.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Audio recording
```
Comment (srietkerk) [inline libs/audio-recording/docs/reference/record.md:1]: Curious why both `audio-recording.md` and `record.md` are needed. This look like similar content to me.

File: libs/audio-recording/docs/reference/record/audio-status.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# audio Status
+
+Check to see if an audio buffer status is true or not.
```
Comment (srietkerk) [inline libs/audio-recording/docs/reference/record/audio-status.md:3]: This is a bit nitpicky, but I get a bit confused when I read "an audio buffer status". 

I'm not sure what would be better though.. I played with "a status for the audio buffer" but I'm not sure if that's better.

File: libs/audio-recording/docs/reference/record/set-sample-rate.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+When playing back, the sample rate sets the speed at which the sound is taken from the audio buffer and sent to the speaker or sound output pin. If the audio in the buffer was recorded at `11000` samples per second but the current sample rate is set at `22000`, the playback of the audio is twice as fast as it was recorded and it will sound different. The sound will play slower that the recorded sound if the playback sample rate is set to `550` before playing the audio back.
+
+## Parameters
+
+* **hz**: the [number](/types/number) of samples per second to record or play back.
+* **scope**: an optional operation scope for the sample rate.
+>* `everything`: (default) both recording and playback are at the same rate.
+>* `playback`: audio playback is at this sample rate.
+> * `recording`: audio recording is at this sample rate.
```
Comment (srietkerk) [inline libs/audio-recording/docs/reference/record/set-sample-rate.md:19]: Being that this block is actively changing the sample rate, I find the "is at" language to be a bit confusing. Maybe `x is changed to this sample rate`?

**Comment by jwunderl** [comment]
@ganicke I'm gonna merge this so I can port to stable so the `//% help=` paths get picked up, so will have to separate out the suggestions from sarah into a separate pr~

**Review by jwunderl — APPROVED** [review]

**Pull request merged**